src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingSpan.cs

```diff
@@ -51,14 +51,9 @@ public int MinCSharpIndentLevel
         {
             get
             {
-                if (IsInClassBody)
-                {
-                    return 2;
-                }
-                else
-                {
-                    return 3 + ComponentLambdaNestingLevel;
-                }
+                var baseIndent = IsInClassBody ? 2 : 3;
+
+                return baseIndent + ComponentLambdaNestingLevel;
```

Comment:
Oh wow so interesting. I dug through some of the history to remember why Ajay built it the other way at first and it may have been to protect against an extreme edgecase where an @code block is nested in a component? 😆  Anyhow, that world is so unlikely, this is simple and correct!

Reply:
Should be easy enough to make the component tracking smarter if thats a scenario, just reset the count to 0 in the formatting visitor (i think??). Do you want me to add a test? Would it just be:
```
<Counter>
	@code
	{
		<Counter>
			// blah
		</Counter>
	}
</Counter>
```

??

Reply:
> Do you want me to add a test?

Honestly it's a supppper edge case I wouldn't stress it. I think I'd be more curious of folks reporting the fact that they did it? lol technically there's no functional difference in having it in vs. outside of the ocmponent.

> Would it just be:

Yup

Reply:
> . I think I'd be more curious of folks reporting the fact that they did it

Good enough for me!